src/OpenApi/gen/XmlComments/MemberKey.cs

```diff
@@ -23,17 +23,68 @@ internal sealed record MemberKey(
 
     public static MemberKey FromMethodSymbol(IMethodSymbol method)
     {
+        string returnType;
+        if (method.ReturnsVoid)
+        {
+            returnType = "typeof(void)";
+        }
+        else
+        {
+            // Handle Task/ValueTask for async methods
+            var actualReturnType = method.ReturnType;
+            if (method.IsAsync && actualReturnType is INamedTypeSymbol namedType)
+            {
+                if (namedType.TypeArguments.Length > 0)
+                {
+                    actualReturnType = namedType.TypeArguments[0];
+                }
+                else
+                {
+                    returnType = "typeof(void)";
+                }
+            }
+
+            returnType = actualReturnType.TypeKind == TypeKind.TypeParameter
+                ? "typeof(object)"
+                : $"typeof({actualReturnType.ToDisplayString(_typeKeyFormat)})";
```

Comment:
Looks like the assignment in the `else` here is redundant as it gets overwritten after the block?